src/dos/cdrom.cpp

```diff
@@ -18,6 +18,217 @@
+                                            unsigned char &attr)
+{
+	if (CD_INDRIVE(SDL_CDStatus(cd))) {
+		start = frames_to_msf(cd->track[track - 1].offset);
+		attr = cd->track[track - 1].type << 4; // sdl uses 0 for audio
+		                                       // and 4 for data. instead
+		                                       // of 0x00 and 0x40
+	}
+	return CD_INDRIVE(SDL_CDStatus(cd));
+}
+
+bool CDROM_Interface_SDL::GetAudioSub(unsigned char &attr,
+                                      unsigned char &track,
+                                      unsigned char &index,
+                                      TMSF &relPos,
+                                      TMSF &absPos)
+{
+	if (CD_INDRIVE(SDL_CDStatus(cd))) {
+		track = cd->cur_track;
+		index = cd->cur_track;
+		attr = cd->track[track].type << 4;
+		relPos = frames_to_msf(cd->cur_frame);
+		absPos = frames_to_msf(cd->cur_frame + cd->track[track].offset);
+	}
+	return CD_INDRIVE(SDL_CDStatus(cd));
+}
+
+bool CDROM_Interface_SDL::GetAudioStatus(bool &playing, bool &pause)
+{
+	if (CD_INDRIVE(SDL_CDStatus(cd))) {
+		playing = (cd->status == CD_PLAYING);
+		pause = (cd->status == CD_PAUSED);
+	}
+	return CD_INDRIVE(SDL_CDStatus(cd));
+}
+
+bool CDROM_Interface_SDL::GetMediaTrayStatus(bool &mediaPresent,
+                                             bool &mediaChanged,
+                                             bool &trayOpen)
+{
+	SDL_CDStatus(cd);
+	mediaPresent = (cd->status != CD_TRAYEMPTY) && (cd->status != CD_ERROR);
+	mediaChanged = (oldLeadOut != cd->track[cd->numtracks].offset);
+	trayOpen = !mediaPresent;
+	oldLeadOut = cd->track[cd->numtracks].offset;
+	if (mediaChanged)
+		SDL_CDStatus(cd);
+	return true;
+}
+
+bool CDROM_Interface_SDL::PlayAudioSector(const uint32_t start, uint32_t len)
+{
+	// Has to be there, otherwise wrong cd status report (dunno why, sdl bug ?)
+	SDL_CDClose(cd);
+	cd = SDL_CDOpen(driveID);
+	bool success = (SDL_CDPlay(cd, start + 150, len) == 0);
+	return success;
+}
+
+bool CDROM_Interface_SDL::PauseAudio(bool resume)
+{
+	bool success;
+	if (resume)
+		success = (SDL_CDResume(cd) == 0);
+	else
+		success = (SDL_CDPause(cd) == 0);
+	return success;
+}
+
+bool CDROM_Interface_SDL::StopAudio(void)
+{
+	// Has to be there, otherwise wrong cd status report (dunno why, sdl bug ?)
+	SDL_CDClose(cd);
+	cd = SDL_CDOpen(driveID);
+	bool success = (SDL_CDStop(cd) == 0);
+	return success;
+}
+
+bool CDROM_Interface_SDL::LoadUnloadMedia([[maybe_unused]] bool unload)
+{
+	bool success = (SDL_CDEject(cd) == 0);
+	return success;
+}
+
+int CDROM_GetMountType(char *path, int forceCD)
+{
+	// 0 - physical CDROM
+	// 1 - Iso file
+	// 2 - subdirectory
+	// 1. Smells like a real cdrom
+	// if ((strlen(path)<=3) && (path[2]=='\\') &&
+	// (strchr(path,'\\')==strrchr(path,'\\')) &&
+	// (GetDriveType(path)==DRIVE_CDROM)) return 0;
+
+	const char *cdName;
+	char buffer[512];
+	strcpy(buffer, path);
+#if defined(WIN32)
+	upcase(buffer);
+#endif
```

Comment:
Again, you can probably use `std::string` here.

Also is it absolutely necessary to restrict the upper case to Windows?

Reply:
Fixed accordingly. Done.